WordPress/gutenberg
# Fix undo/redo in site editor code editor's mode

closes #43277 

## What?

This PR fixes the undo/redo of the site editor's code editor mode but also simplifies the code editor a lot. 

## How?

The main complexity in the previous implementation is that it was actually making the assumption that when you edit the "content" of a record, you also need to edit the "blocks" property to keep it in sync. The reality is that we don't have to keep these two in sync, we can instead just "unset" the blocks which means whenever the blocks will be needed again, they'll just parse again from the edited "content".

Ideally this PR is also merged in coordination with #52417 which simplifies the blocks parsing logic from content too. 

I think we can probably do the same simplification to the code editor of edit-post too.

## Testing Instructions

1- Open the site editor 
2- Switch to code editor mode
3- Edit the content and use undo/redo
4- Everything should work perfectly (undo/redo is per character but that's a separate issue anyway).
